Kafka log plugin (#1005)
Kong/developer.konghq.com: app/_kong_plugins/kafka-log/index.md

  - kafka-log
---

Publish request and response logs to an [Apache Kafka](https://kafka.apache.org/) topic. This plugin does not support message compression.
For more information, see [Kafka topics](https://kafka.apache.org/documentation/#intro_concepts_and_terms).

Kong also provides a Kafka plugin for request transformations. See [Kafka Upstream](/plugins/kafka-upstream/).

## Log format

{:.info}
> **Note:** If the `max_batch_size` argument > 1, a request is logged as an array of JSON objects.

{% include /plugins/logging/log-format.md %}

## Implementation details

This plugin uses the [lua-resty-kafka](https://github.com/kong/lua-resty-kafka) client.

When encoding request bodies, several things happen:

* For requests with a content-type header of `application/x-www-form-urlencoded`, `multipart/form-data`,
  or `application/json`, this plugin passes the raw request body in the `body` attribute, and tries

>   to return a parsed version of those arguments in `body_args`. If this parsing fails, an error message is
>   returned and the message is not sent.

Reviewer: ```suggestion
  to return a parsed version of those arguments in `body_args`. 
  If this parsing fails, the plugin returns an error message and the message isn't sent.
```

* If the `content-type` is not `text/plain`, `text/html`, `application/xml`, `text/xml`, or `application/soap+xml`,

>   then the body will be base64-encoded to ensure that the message can be sent as JSON. In such a case,

Reviewer: ```suggestion
  then the body will be base64-encoded to ensure that the message can be sent as JSON. In that case,
```
